cmd/ltctl/comparison.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/mattermost/mattermost-load-test-ng/comparison"
+
+	"github.com/spf13/cobra"
+)
+
+func processResults(results []comparison.Result) error {
+	for i, res := range results {
+		fmt.Println("==================================================")
+		fmt.Println("Comparison result:")
+		if res.Report != "" {
+			filename := fmt.Sprintf("report_%d.md", i)
+			if err := ioutil.WriteFile(filename, []byte(res.Report), 0660); err != nil {
+				return err
+			}
+			fmt.Printf("Report: %s\n", filename)
+		}
+		if res.DashboardURL != "" {
+			fmt.Printf("Grafana Dashboard: %s\n", res.DashboardURL)
+		}
+		for _, ltRes := range res.LoadTests {
+			fmt.Printf("%s:\n", ltRes.Label)
+			fmt.Printf("  Type: %s\n", ltRes.Config.Type)
+			fmt.Printf("  DB Engine: %s\n", ltRes.Config.DBEngine)
+			if ltRes.Config.Type == comparison.LoadTestTypeBounded {
+				fmt.Printf("  Duration: %s\n", ltRes.Config.Duration)
+				fmt.Printf("  Users: %d\n", ltRes.Config.NumUsers)
+			} else if ltRes.Config.Type == comparison.LoadTestTypeUnbounded {
+				fmt.Printf("  Supported Users: %d\n", ltRes.Status.SupportedUsers)
+			}
+			fmt.Printf("  Errors: %d\n", ltRes.Status.NumErrors)
+		}
+		fmt.Printf("==================================================\n\n")
+	}
+	return nil
+}
+
+func RunComparisonCmdF(cmd *cobra.Command, args []string) error {
+	deployerConfig, err := getConfig(cmd)
```

Comment:
What do you mean by "adding another app"? The deployer config in this case is used more like a template that gets replicated for all needed deployments with only a couple of settings dynamically changed in code (e.g. database type). 

This is done to both abstract a lot of the complexity away from the user and simplify the overall process. I felt that having a separate deployer config per needed deployment was too much of a customization (and overhead) with in the reality of comparisons is hardly needed.

Reply:
Like comparing a feature branch includes an integration with 3rp party app (e.g. ES). Even if we use different deployer configs probably we will need to update the code as well. So on daily basis it wouldn't be straightforward. I just wanted to hear your thoughts.

Reply:
If the "new" app is not covered by the load-test tool that's a different issue. We would have to add code for it first and make the necessary changes so that it gets deployed/provisioned (like we did for ES). 

For example, we want to add support some basic LDAP functionality. We need to implement the code to trigger that (e.g. login) and also either provision an LDAP server in some instance we have or create a new one for it. There's also another option which is to leave that configurable so that an external (already existing) LDAP server could be provided and no deployment needed. Either way some change in the code has to be made.

Is your worry that the `deployer.Config` object will become increasingly complex?

Reply:
This was further discussed "offline". The conclusion is that there are cases where we might want to test two builds making use of different services (e.g. two competing search engines). These would require different deployments hence @isacikgoz's concern.
The idea here is to keep things simple and handle these less common cases by allowing users to specify existing app servers/clusters to run the comparison on. This is already planned and will be covered by [MM-30398](https://mattermost.atlassian.net/browse/MM-30398).